accounts-db/src/accounts_db.rs

```diff
@@ -1349,16 +1351,55 @@ impl StoreAccountsTiming {
 struct CleaningInfo {
     slot_list: SlotList<AccountInfo>,
     ref_count: u64,
-    /// True for pubkeys mapping to older versions of accounts that should be purged.
-    should_purge: bool,
+}
+
+#[derive(Debug)]
+struct CleaningMap {
+    bin_calculator: PubkeyBinCalculator24,
+    bins: Box<[RwLock<HashMap<Pubkey, CleaningInfo>>]>,
```

Comment:
Initially, I wanted to store a reference to `PubkeyBinCalculator24` in the field, but got entangled in lifetime parameters. Since `PubkeyBinCalculator24` contains a single u32 value, I thought it might be acceptable to clone it.